src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -212,7 +212,7 @@ private void SetupAuthorities(string aadAuthority, ProtocolVersion version)
         /// <exception cref="ArgumentNullException"> if <paramref name="securityToken"/> is null.</exception>
         /// <exception cref="ArgumentNullException"> if <paramref name="validationParameters"/> is null.</exception>
         /// <exception cref="SecurityTokenInvalidIssuerException">if the issuer is invalid or if there is a network issue. </exception>
-        internal async ValueTask<string> ValidateAsync(
+        public async ValueTask<string> ValidateAsync(
```

Comment:
Should we still return a ValueTask here? Now ValidateAsync is mostly sync, except making the async call to get the configuration. 

My understanding ValueTask has limitations like they can only be awaited once unlike Task. Perhaps more general Task<string> return type is more appropriate for a public API?

(Later if we do decide to deprecate this method later, and have one that returns a Task, it be a bit difficult since return type is not part of the method signature).

Reply:
That is a good point, taking this into consideration I think we should not haste this part. I will make the internals visible to the IdentityWeb instead. Thanks Peter